File: training/medium/conway-sequence.js
var R = parseInt(readline());
var L = parseInt(readline());

function conway(t) {
    r = [];
    idx = 0;
    while (idx < t.length){
        for(i=1; t[idx+i] == t[idx]; i++) {}
        r.push(i);
        r.push(t[idx]);
        idx += i;
    }
    return r;
}

value = [R];
var j = 1;

while (j<L){
    value = conway(value);
    j=j+1;
}

print(value.join(' '));

File: training/medium/conway-sequence.py
import sys
from itertools import groupby

def conway(suite):
    next_terme = []
    for k, g in groupby(suite):
        next_terme.append(str(len(list(g))))
        next_terme.append(k)
    return next_terme
    

r = int(input()) # initial number
l = int(input()) # number of step

terme = [str(r)]
freq = []
key = []

for i in range(1, l):
    terme = conway(terme)
    print(i, terme, file=sys.stderr)

print(' '.join(terme))
# To debug: print("Debug messages...", file=sys.stderr)
   
